cartridges/int_algolia/cartridge/scripts/algolia/model/algoliaLocalizedProduct.js

```diff
@@ -359,9 +360,18 @@ function algoliaLocalizedProduct(parameters) {
 
         for (var i = 0; i < attributeList.length; i += 1) {
             var attributeName = attributeList[i];
+            var attributeNameObj = attributeName.split('.');
+            if (attributeNameObj.length > 1) {
+                var parentAttribute = attributeNameObj[0];
+                var subAttribute = attributeNameObj[1];
+            }
 
             if (baseModel && baseModel[attributeName]) {
                 this[attributeName] = baseModel[attributeName];
+            } else if (baseModel && parentAttribute && subAttribute && baseModel[parentAttribute] && baseModel[parentAttribute][subAttribute]) {
```

Comment:
To make things more generic, it's probably possible to do something similar than the `ObjectHelper.getAttributeValue()` method (using `reduce`):

```js
function getAttributeValue(attributeNameArr, model) {
  return attributeNameArr.reduce((prev, attr) => {
    if (!prev || !prev[attr]) {
      return null;
    }
    return prev[attr];
  }, model);
}
```

And do the same for writing.
But if it's too complex, it's ok like that.